examples/parametric_cart_pole_with_friction.py

```diff
@@ -0,0 +1,136 @@
+import numpy as np
+from casadi import SX, horzcat, vertcat, cos, sin, inv
+import matplotlib.pyplot as plt
+
+import nosnoc
+
+# TODO:% 2 ) make the masses m_1 and m_2 parameteres (constant over time)
+
+PARAMETRIC = 1
+def main():
+    # opts
+    opts = nosnoc.NosnocOpts()
+    opts.irk_scheme = nosnoc.IrkSchemes.RADAU_IIA
+    opts.n_s = 2
+    opts.homotopy_update_rule = nosnoc.HomotopyUpdateRule.SUPERLINEAR
+    opts.step_equilibration = nosnoc.StepEquilibrationMode.HEURISTIC_MEAN
+    opts.equidistant_control_grid = True
+
+    opts.N_stages = 50  # number of control intervals
+    opts.N_finite_elements = 2  # number of finite element on every control intevral
+    opts.terminal_time = 4.0  # Time horizon
+    opts.print_level = 1
+    # faster options
+    opts.N_stages = 15  # number of control intervals
+
+    ## Model defintion
+    q = SX.sym('q', 2)
+    v = SX.sym('v', 2)
+    x = vertcat(q, v)
+    u = SX.sym('u')  # control
+
+    ## parametric version:
+    # masses
+    m1 = SX.sym('m1')  # cart
+    m2 = SX.sym('m2')  # link
+    x_ref = SX.sym('x_ref', 4)
+    p = vertcat(m1, m2, x_ref)
+
+    x_ref_val = np.array([0, 180 / 180 * np.pi, 0, 0])  # end upwards
+
+    # either provide single parameter vector
+    p_val = np.concatenate((np.array([1.0, 0.1]), x_ref_val))
+    # or matrix (one parameter vector for each control stage)
+    p_val = np.tile(p_val, (opts.N_stages, 1))
```

Comment:
Do we really need to copy the parameter values over all stages manually?
After some thinking, I would still prefer  say `p_global_val, p_time_var_val`
and `p_global` and `p_time_var`  over `p`
(and, of course, put them together and copy the values in the solver construction)